segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/stat/SegmentStatsRecorderTest.java

```diff
@@ -181,6 +181,28 @@ public void testMetrics() {
         assertEquals(NO_COUNTER_VALUE, getCounterValue(MetricsNames.SEGMENT_READ_BYTES, segmentName));
     }
 
+    @Test(timeout = 10000)
+    public void testTransactionMetrics() throws InterruptedException {
+        val segmentName = getStreamSegmentName();
+
+        long dataLength = 321;
+        val duration = Duration.ofMinutes(1);
+        @Cleanup
+        val context = new TestContext(segmentName, duration, false);
+
+        context.statsRecorder.createSegment(segmentName, ScalingPolicy.ScaleType.BY_RATE_IN_KBYTES_PER_SEC.getValue(), 2, Duration.ofSeconds(1));
+        context.statsRecorder.recordAppend(segmentName, dataLength, 5, Duration.ofSeconds(2));
+        val txnName1 = NameUtils.getTransactionNameFromId(segmentName, UUID.randomUUID());
+        context.statsRecorder.createSegment(txnName1, ScalingPolicy.ScaleType.BY_RATE_IN_KBYTES_PER_SEC.getValue(), 2, Duration.ofSeconds(1));
+        // Make sure deletions do not cause side effects.
+        context.statsRecorder.deleteSegment(txnName1);
+        assertEquals(dataLength, getCounterValue(SEGMENT_WRITE_BYTES, segmentName));
+        // All closures of metrics happen through the SimpleCache. Asserting that an entry for `txnName1` does not exist
+        // ensures that there is guaranteed to be a one-one mapping of entries to counters.
+        assertNull(context.statsRecorder.getSimpleCache().get(txnName1));
```

Comment:
You can use `context.statsRecorder.getSegmentAggregates(txnName1)` so you don't have to expose the cache.